aten/src/ATen/native/cuda/ScanKernels.cu

```diff
@@ -471,14 +477,54 @@ void scan_thrust(const Tensor& self, Tensor& result, scalar_t init, BinaryFuncti
       src_data, src_data + size, dst_data,
       rocm_bug_binary_op);
   #else
-  thrust::device_ptr<scalar_t> src_data(self.data_ptr<scalar_t>());
-  thrust::device_ptr<scalar_t> dst_data(result.data_ptr<scalar_t>());
-  ptrdiff_t size = self.numel();
-  thrust::inclusive_scan(
-      thrust::cuda::par(allocator).on(c10::cuda::getCurrentCUDAStream()),
-      src_data, src_data + size, dst_data,
-      binary_op);
-  #endif
+  int64_t size = self.numel();
+  // non synchronizing cub call
```

Comment:
Should we rename the function `scan_thrust` to a different name?

Reply:
Done. Don't want to deal with rocm for now, so they remain on thrust.